### ISSUE TITLE
[JENKINS-7004] cause not working in parameterized projects

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-14#8179]
<p>The documentation in <em>trigger builds remotely</em> says</p>

<blockquote><p>Optionally append <tt>&cause=Cause+Text</tt> to provide text that will be included in the recorded build cause.</p></blockquote>

<p>The <tt>cause</tt> doesn't work for parameterized builds.  </p>

<p>Also the documentation should state that the trigger URL in that case has <tt>buildWithParameters</tt> not <tt>build</tt>...</p>

<p>Already requested in <a href="http://wiki.jenkins-ci.org/display/JENKINS/Parameterized+Build?focusedCommentId=40042520#comment-40042520" class="external-link" target="_blank" rel="nofollow noopener">wiki</a></p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=anshnd">anshnd</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-7004" target="_blank">cause not working in parameterized projects</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2010-07-18T08:59:03+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-11-14
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>The documentation in <em>trigger builds remotely</em> says</p>

<blockquote><p>Optionally append <tt>&amp;cause=Cause+Text</tt> to provide text that will be included in the recorded build cause.</p></blockquote>

<p>The <tt>cause</tt> doesn't work for parameterized builds.  </p>

<p>Also the documentation should state that the trigger URL in that case has <tt>buildWithParameters</tt> not <tt>build</tt>...</p>

<p>Already requested in <a href="http://wiki.jenkins-ci.org/display/JENKINS/Parameterized+Build?focusedCommentId=40042520#comment-40042520" class="external-link" target="_blank" rel="nofollow noopener">wiki</a></p></pre>
</details>
</details>
<details><summary><i>1 attachment</i></summary>

- [jenkinstestharness.txt](https://issues.jenkins.io/secure/attachment/46342/jenkinstestharness.txt)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link's <a href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=140245&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-140245">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Code changed in hudson
User: : mindless
Path:
 trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java
 trunk/hudson/main/core/src/main/java/hudson/model/ParametersDefinitionProperty.java
 trunk/hudson/main/core/src/main/resources/hudson/model/BuildAuthorizationToken/config.jelly
 trunk/www/changelog.html
<a href="http://jenkins-ci.org/commit/33032" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/33032</a>
Log:
<span class="error">&#91;FIXED JENKINS-7004&#93;</span> Make /buildWithParameters support remote cause and
user supplied cause text for build via authentication token, just as /build does.</p>
</pre>
</details>

</details>
<p>Code changed in hudson<br/>
User: : mindless<br/>
Path:<br/>
 trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java<br/>
 trunk/hudson/main/core/src/main/java/hudson/model/ParametersDefinitionProperty.java<br/>
 trunk/hudson/main/core/src/main/resources/hudson/model/BuildAuthorizationToken/config.jelly<br/>
 trunk/www/changelog.html<br/>
<a href="http://jenkins-ci.org/commit/33032" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/33032</a><br/>
Log:<br/>
<span class="error">&#91;FIXED JENKINS-7004&#93;</span> Make /buildWithParameters support remote cause and<br/>
user supplied cause text for build via authentication token, just as /build does.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=362409&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362409">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<blockquote><p>The job MUST have a `token` assigned</p></blockquote>
<p>This feature only exists for triggering builds via token and nowhere else, which is why it is part of that feature's documentation. Adding it elsewhere would be a feature request.</p>
</pre>
</details>

</details>
<blockquote><p>The job MUST have a `token` assigned</p></blockquote>
<p>This feature only exists for triggering builds via token and nowhere else, which is why it is part of that feature's documentation. Adding it elsewhere would be a feature request.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cvijayan's <a href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=362613&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362613">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>My intent is to figure out if there is a way to monitor a job invocation accurately by either using the Queue ID returned as part of the Location and then retrieving the Build ID  OR by passing a unique identifier to the Cause parameter.</p>

<p>What I find weird is that when there is a Quiet period in effect neither the Queue ID nor the Cause parameter behave as expected. Please see the attached sample test harness and its output. </p>

<p>There are 3 jobs invoked in parallel each a unique identifier for cause. This is is the outcome.</p>

<p>A. First job gets a successful build, the other 2 don't (presumably because of the Quiet period).</p>

<p>B.  The First job is queryable through the Queue ID and the response does have the unique identifier passed in Cause. </p>

<p>C. However the second and third invocations get a status of 201 and have the same Queue ID of the first, with the response having all the 3 causes. </p>

<p> </p>

<p>  </p>
</pre>
</details>

</details>
<p>My intent is to figure out if there is a way to monitor a job invocation accurately by either using the Queue ID returned as part of the Location and then retrieving the Build ID  OR by passing a unique identifier to the Cause parameter.</p>

<p>What I find weird is that when there is a Quiet period in effect neither the Queue ID nor the Cause parameter behave as expected. Please see the attached sample test harness and its output. </p>

<p>There are 3 jobs invoked in parallel each a unique identifier for cause. This is is the outcome.</p>

<p>A. First job gets a successful build, the other 2 don't (presumably because of the Quiet period).</p>

<p>B.  The First job is queryable through the Queue ID and the response does have the unique identifier passed in Cause. </p>

<p>C. However the second and third invocations get a status of 201 and have the same Queue ID of the first, with the response having all the 3 causes. </p>

<p> </p>

<p>  </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cvijayan's <a href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=362614&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362614">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Should I open a new ticket or reopen this ticket  ?</p>
</pre>
</details>

</details>
<p>Should I open a new ticket or reopen this ticket  ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=362618&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362618">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>The behavior described above is as designed, queue items are combined when they're considered identical, and causes combined as well. So if you pass the same build parameters (the Magic "cause" is not one of them) they will be combined.</p>
</pre>
</details>

</details>
<p>The behavior described above is as designed, queue items are combined when they're considered identical, and causes combined as well. So if you pass the same build parameters (the Magic "cause" is not one of them) they will be combined.</p>
